test/basic.bats

```diff
@@ -69,7 +69,8 @@ function teardown() {
     [ "$(cat oci/blobs/sha256/$config | jq -r '.config.Entrypoint | join(" ")')" = "echo hello world" ]
 
     stackerGit=$(cat oci/blobs/sha256/$manifest | jq -r '.annotations."ws.tycho.stacker.git_version"')
-    [ "stacker version ${stackerGit}" = "$(stacker --version)" ]
+    rootDir=$(git rev-parse --show-toplevel)
+    [ "stacker version ${stackerGit}" = "$(${rootDir}/stacker --version)" ]
```

Comment:
Can you describe what was broken here? AFAIK this should be a no-op.

Reply:
It was using the stacker in the PATH instead of the stacker in the local folder. So the versions were different.

Reply:
It should really be using the function "stacker" from test/helpers.bash; can you describe how to reproduce it? This has worked for me for quite a while...

Reply:
Does `$()` for with functions?

Reply:
```
~ function foo() { echo "hello world"; } 
~ echo "$(foo)"
hello world
```

Yep. Can you describe how to reproduce the problem you were seeing?

Reply:
```
andaaron@andaaron-ThinkPad-P51s:~/anuvu/stacker$ sudo bats test/basic.bats --tap
[sudo] password for andaaron: 
1..1
not ok 1 basic workings
# (in test file test/basic.bats, line 73)
#   `[ "stacker version ${stackerGit}" = "$(stacker --version)" ]' failed
# Getting image source signatures
# Copying blob sha256:8ba884070f611d31cb2c42eddb691319dc9facf5e0ec67672fcfa135181ab3df
# Copying config sha256:b9a1b1f0b2baaec83946a26d7045e4028f11eccc8b0e5b3514568e56a391beb2
# Writing manifest to image destination
# Storing signatures
# initializing stacker recipe: stacker.yaml
# substituting $FAVICON to favicon.ico
# stacker build order:
# 0 build /home/andaaron/anuvu/stacker/test/stacker.yaml: requires: []
# building: 0 /home/andaaron/anuvu/stacker/test/stacker.yaml
# substituting $FAVICON to favicon.ico
# building image centos...
# importing files...
# copying /home/andaaron/anuvu/stacker/test/stacker.yaml
# downloading https://www.cisco.com/favicon.ico
 1.37 KiB / 1.37 KiB [========================================================] 100.00% 2.03 MiB/s 0s
# copying /home/andaaron/anuvu/stacker/test/executable
# using cached copy of .stacker/layer-bases/centos.tar
# running commands...
# running commands for centos
# + cp /stacker/favicon.ico /favicon.ico
# + cp /stacker/executable /usr/bin/executable
# generating layer for centos
# setting git version annotation to 7aff18162f3533559a6b466f10d6f2faa71da6eb-dirty
# filesystem centos built successfully
# building image layer1...
# importing files...
# running commands...
# running commands for layer1
# + rm /favicon.ico
# generating layer for layer1
# setting git version annotation to 7aff18162f3533559a6b466f10d6f2faa71da6eb-dirty
# filesystem layer1 built successfully
andaaron@andaaron-ThinkPad-P51s:~/anuvu/stacker$ stacker --version
stacker version e5283a6aceab5f3ca795bf80c2399d8c91f5a8b0-dirty
andaaron@andaaron-ThinkPad-P51s:~/anuvu/stacker$ ./stacker --version
stacker version 7aff18162f3533559a6b466f10d6f2faa71da6eb
andaaron@andaaron-ThinkPad-P51s:~/anuvu/stacker$ git rev-parse HEAD
7aff18162f3533559a6b466f10d6f2faa71da6eb
```

Reply:
Nevermind, it was because I didn't rebuild stacker. And the new binary should have included 'dirty' in the name.

Reply:
@tych0 so should the version in the manifest be the version of the stacker binary used for building?
Or should it be the version of the repo where the stacker yaml is located?

Reply:
Removed this commit

Reply:
It's the repo where the stacker.yaml is located; they just happen to be the same for the actual stacker repo.